tests/src/lib.rs

```diff
@@ -211,6 +211,22 @@ mod tests {
         roundtrip::<foo::bar_baz::FooBarBaz>(&buf).unwrap();
     }
 
+    #[test]
+    fn test_custom_container_attributes() {
```

Comment:
Good test, but please extend it to cover enums and one-ofs.  In particular I'm not sure how the current code will work with one-ofs, since they don't really have a Protobuf type name.